hs_file_types/models/reftimeseries.py

```diff
@@ -709,21 +817,25 @@ def get_copy(self):
 
 def _extract_metadata(resource, logical_file):
     # add resource level title if necessary
-    if resource.metadata.title.value == 'Untitled resource':
+    if resource.metadata.title.value.lower() == 'untitled resource' \
+            and logical_file.metadata.has_title_in_json:
         resource.metadata.update_element('title', resource.metadata.title.id,
-                                         value=logical_file.metadata.title)
+                                         value=logical_file.dataset_name)
 
-    # add resource level abstract id necessary
-    if resource.metadata.description is None:
-        resource.metadata.create_element('description', abstract=logical_file.metadata.abstract)
+    # add resource level abstract if necessary
+    if resource.metadata.description is None and logical_file.metadata.has_abstract_in_json:
+        resource.metadata.create_element('description',
+                                         abstract=logical_file.metadata.get_abstract_from_json())
     # add resource level keywords
-    resource_keywords = [kw.value.lower() for kw in resource.metadata.subjects.all()]
-    for kw in logical_file.metadata.key_words:
-        if kw.lower() not in resource_keywords:
-            resource.metadata.create_element('subject', value=kw)
+    if logical_file.metadata.has_keywords_in_json:
+        resource_keywords = [kw.value.lower() for kw in resource.metadata.subjects.all()]
+        for kw in logical_file.metadata.get_keywords_from_json():
+            if kw.lower() not in resource_keywords:
+                resource.metadata.create_element('subject', value=kw)
 
     # add to the file level metadata
-    logical_file.metadata.keywords = logical_file.metadata.key_words
+    logical_file.metadata.keywords = logical_file.metadata.get_keywords_from_json()
```

Comment:
@pkdash It'd be good to save keywords retrieved by calling ```get_keywords_from_json``` in a variable and use it subsequently in the same method rather than calling it again.